servers/quarkus-server/src/main/resources/application.properties

```diff
@@ -110,7 +110,8 @@ quarkus.resteasy.gzip.enabled=true
 #quarkus.oidc.auth-server-url=
 # fixed at buildtime
 quarkus.http.auth.basic=false
-quarkus.oidc.enabled=false
+quarkus.oidc.enabled=true
```

Comment:
Not actually a fan of enabling this by default. I suspect, that all requests even for a "try it out w/ docker" scenario would require OIDC, then?

Reply:
For test classes, you can use a custom quarkus test profile:
* a class that extends QuarkusTestProfile - one for native tests and one for non-native tests
* annotate the tests using `@io.quarkus.test.junit.TestProfile`

Reply:
Do we not want to have OIDC enabled by default? I though that was the intent. It has to be enabled at build time if it is to be usable. 

I believe we can have it enabled but not used in OSS docker images by default. As far as I can tell it should not break anything. Users can specify a keycloak URL at run time as a parameter if they want to integrate with it (this part I have not done yet).

 @rymurr WDYT?

Reply:
The ideal scenario is that OIDC is off by default but is easily activated. We want to cater for:

1. easy to try out and play w/ Nessie docker image w/o building an auth mechanism
2. being able to integrate the docker image into a prod setup w/ OIDC enabled

Is that possible? By the sounds of it you are saying auth is off unless config parameters are set at runtime to enable oidc explicitly? to me 1 is more important than 2 for the base image 

An aside, did you test anything other than keycloak? I believe there were some issues w/ other providers...

Reply:
> you are saying auth is off unless config parameters are set at runtime to enable oidc explicitly?

That's my understanding of Quarkus docs. `quarkus.oidc.enabled` is one of a few settings that must be set at build time. I'll try and confirm manually.

https://quarkus.io/guides/security-openid-connect#configuring-using-the-application-properties-file (`quarkus.oidc.enabled` is marked with a lock sign).

Reply:
> did you test anything other than keycloak? I believe there were some issues w/ other providers...

So far I testing only with keycloak. Will test others too.

Reply:
Confirmed: `quarkus.oidc.enabled` _cannot_ be overwritten to `true` at runtime.

```
2021-08-18 11:04:30,022 WARN  [io.qua.run.con.ConfigChangeRecorder] (main) Build time property cannot be changed at runtime:
 - quarkus.oidc.enabled was 'false' at build time and is now 'true'
```

I'll see if it is possible to have reasonable defaults with OIDC enabled to avoid 500 errors in case a user tries `Bearer` auth without properly configuring the server.